File: Programmers/pro_42883_큰수만들기.py
def solution(number, k):
    idx=0
    N=len(number)
    answer=''
    while(k and idx+k<N):
        cur = number[idx]
        if(cur!='9'):            
            for i in range(idx+1,idx+1+k):
                if(cur<number[i]):
                    k-=1
                    cur=''
                    break
        answer+=cur
        idx+=1
    answer += number[idx:N-k]
    return answer

File: Programmers/pro_42883_큰수만들기.js
function solution(number, k) {
  const answer = [];
  [...number].forEach(num => {
    while (k && answer.length && answer[answer.length - 1] < num) {
      answer.pop()
      k--;
    }
    answer.push(num)
  })
  if (k) answer.splice(-k)
  return answer.join('');
}